dds/idl/marshal_generator.cpp

```diff
@@ -1428,12 +1431,14 @@ namespace {
         be_global->impl_ <<
           indent << "}\n";
       }
-      be_global->impl_ <<
-        "  if (discard_flag) {\n"
-        "    strm.set_construction_status(Serializer::ElementConstructionFailure);\n"
-        "    return false;\n"
-        "  }\n"
-        "  return true;\n";
+	  if (!primitive) {
+          be_global->impl_ <<
```

Comment:
This can be simplified by merging this whole statement with the one that ends with `indent << "}\n";` above since it's the in the `else` block of `if (primitive)`.

Reply:
Good suggestion.  Should be fixed now.